### PR TITLE
drivers: add cc2420 support (based on #5314)

### PR DESCRIPTION
**Pull request by kYc0o**
This is a reworked version of #5314 with code reorganisation and netdev2 compliant.

The code was tested in a z1 board, please feel free to test it on telosb and wsn430-v1_4.

Communication works for L2 messages with short and long addresses, but IPv6 messages are discarded, so that's why it's WIP.

One major bug is that when `dev->netdev.proto = GNRC_NETTYPE_SIXLOWPAN;` is enabled random messages are received from random kernel PIDs.

### CODE REVIEW

**Comment by kYc0o** [comment]
IPv6 messages are being received now, so no more WIP. However the spurious messages still coming, but apparently they don't bother other threads.

**Comment by OlegHahm** [comment]
I tried with a Z1 and a SAMR21-Xpro. Link layer communication (unicast and broadcast) seems to work fine in both directions. :+1: 

However, trying to ping the Z1 running `gnrc_minimal` from SAMR21 with `gnrc_networking` doesn't work. Also, the L2 address doesn't seem unique, but probably the MSP430 doesn't provide CPUID, right?

**Comment by kYc0o** [comment]
After testing on wsn430-1_4 board I found that the UART0 and SPI share the same buffer, thus it conflicts while initialising the driver.  An issue #5599 is open for this

**Comment by kYc0o** [comment]
In order to ping devices using `gnrc_minimal` you need to disable IPHC with something like `ifconfig 7 -iphc`.

For the address yes you're right, MSP430 does not provide CPUID.

**Comment by OlegHahm** [comment]
Ah, I should have known this myself. :)

Confirmed working.

**Comment by kYc0o** [comment]
and ACK :P ?

**Comment by OlegHahm** [comment]
Lemmy quickly look at the code.

File: drivers/Makefile.dep [hunk 1]
```diff
@@ -32,6 +32,21 @@ ifneq (,$(filter cc110x,$(USEMODULE)))
   endif
 endif
 
+ifneq (,$(filter cc2420,$(USEMODULE)))
+  USEMODULE += cc2420
```
Comment (OlegHahm) [inline drivers/Makefile.dep]: Why is this required?

Comment (kYc0o) [inline drivers/Makefile.dep]: Oh yeah, it was also used with `at86rf2xx` so I added it, but you're right, it's not required. Should I remove it?

**Comment by OlegHahm** [comment]
Shouldn't be https://github.com/RIOT-OS/RIOT/pull/5591/commits/ece96d70ad8b462874dd9d5acae30eaedf0ca8f1 and https://github.com/RIOT-OS/RIOT/pull/5591/commits/b2def4a5aebc63e5dcda6a856f69d7102bfac985 be squashed?

**Comment by OlegHahm** [comment]
Apart from the minor comments: ACK.

**Comment by kYc0o** [comment]
Yes I can squash those commits, remove `wsn430-v1_4` and add it when #5599 is solved.

**Comment by kYc0o** [comment]
Squashed and waiting for Murdock. I removed the `USEMODULE += cc2420` can you check once more if everything works fine?

File: drivers/Makefile.dep (hunk 1 above)
Comment (gebart) [inline drivers/Makefile.dep]: The reason for the at86rf2xx USEMODULE check is that the driver is checking
for a wildcard and then adding the generic driver if it matches. The
at86rf2xx driver supports the rf231, rf233, rf212b, (possibly more), and is
written in a way that it relies on USEMODULE to tell it which hardware it
should expect.
Den 5 jul 2016 12:33 skrev "kYc0o" <notifications@github.com>:

In drivers/Makefile.dep
<https://github.com/RIOT-OS/RIOT/pull/5591#discussion_r69541037>:

> @@ -32,6 +32,21 @@ ifneq (,$(filter cc110x,$(USEMODULE)))
>    endif
>  endif
>
> +ifneq (,$(filter cc2420,$(USEMODULE)))
> +  USEMODULE += cc2420

Oh yeah, it was also used with at86rf2xx so I added it, but you're right,
it's not required. Should I remove it?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/RIOT-OS/RIOT/pull/5591/files/ffd03d77950ecaa2210bc2efb066a12ad9728cfd#r69541037>,
or mute the thread
<https://github.com/notifications/unsubscribe/AATYQl9hCQ81LrCAtqWHIs1iLu4Rowqzks5qSjL7gaJpZM4JDOnR>
.

Comment (kYc0o) [inline drivers/Makefile.dep]: Yeah once I analysed I saw that, thanks for the explanation. Now `USEMODULE += cc2420` it's removed.

**Comment by OlegHahm** [comment]
Confirmed, still working. ACK holds.

**Comment by kYc0o** [comment]
Murdock failed because telosb wasn't in some memory insufficient lists for some tests. Fixed and Murdock restarted.

**Comment by emmanuelsearch** [comment]
Murdock has failed. Not sure why though. Any clues?

**Comment by kYc0o** [comment]
Yes it's the doc, I made a `make doc` and corrected some warnings but there still are some,  I can fix it now.

**Comment by kYc0o** [comment]
Hoping this time will make it...

**Comment by emmanuelsearch** [comment]
it seems it failed again?

**Comment by kYc0o** [comment]
mmm lots of warnings are thrown because of doxygen but actually is every word that is commented... very weird...

**Comment by emmanuelsearch** [comment]
so what's the plan?

File: drivers/include/cc2420.h [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+} cc2420_params_t;
+/** @} */
+
+/**
+ * @brief   Device descriptor for CC2420 radio devices
+ * @{
+ */
+typedef struct {
+    /* netdev fields */
+    netdev2_ieee802154_t netdev;
```
Comment (thomaseichinger) [inline drivers/include/cc2420.h]: @kYc0o Running locally I get a warning for this member not being documented.

Comment (kYc0o) [inline drivers/include/cc2420.h]: Yes me too and I fixed it but I'm also having this warnings:
```
warning: ignoring unknown tag `Copyright' at line 2, file cc2420.h
warning: ignoring unknown tag `C' at line 2, file cc2420.h
warning: ignoring unknown tag `2015' at line 2, file cc2420.h
warning: ignoring unknown tag `Freie' at line 2, file cc2420.h
warning: ignoring unknown tag `Universit' at line 2, file cc2420.h
warning: ignoring unknown tag `t' at line 2, file cc2420.h
warning: ignoring unknown tag `Berlin' at line 2, file cc2420.h
warning: ignoring unknown tag `This' at line 4, file cc2420.h
warning: ignoring unknown tag `file' at line 4, file cc2420.h
warning: ignoring unknown tag `is' at line 4, file cc2420.h
warning: ignoring unknown tag `subject' at line 4, file cc2420.h
warning: ignoring unknown tag `to' at line 4, file cc2420.h
```
and so on... I'll push it again and see if that was the warning Murdock is complaining on...

**Comment by kYc0o** [comment]
Now it complains on something completely different. I'll restart it but it seems that the warnings on doxygen are not there anymore...

**Comment by thomaseichinger** [comment]
What was it complaining about?

**Comment by kYc0o** [comment]
Sorry, I restarted it too quickly... It was about the cortex_m3 family, there was "no more space left on device" while compiling some unit tests.

**Comment by kYc0o** [comment]
Same error again... @kaspar030 can you take a quick look into that? I'm afraid there's really no more space left on Murdock or something similar...

**Comment by kaspar030** [comment]
hmm. murdock still has almost 16gig on that drive. I'll watch it a little.

**Comment by kaspar030** [comment]
Yep, the ramdisk is getting full. Every package get's checked out ~40 times, so if more than four tests using packages get's built, there's no space left. I'll think of something.

**Comment by kYc0o** [comment]
Ups... sorry for break Murdock, if I can do something on it just let me know  :)

**Comment by kaspar030** [comment]
@kYc0o please help quicktesting #5610. :)

**Comment by kYc0o** [comment]
Great! doing it now!

**Comment by kYc0o** [comment]
Let's see if #5610 works ;)

**Comment by kYc0o** [comment]
Yeeeeeah!!! Murdock rules! Maybe it's needed to squash the doc fix?

**Comment by OlegHahm** [comment]
Yes, please squash and then merge as soon as Murdock confirms again.

**Comment by kYc0o** [comment]
And GO!
